pkg/services/app/app.go

```diff
@@ -49,55 +48,30 @@ type AppService interface {
 }
 
 type App struct {
-	Osys               osys.Osys
-	Git                git.Git
-	Flux               flux.Flux
-	Kube               kube.Kube
-	Logger             logger.Logger
-	GitProviderFactory func(token string) (gitproviders.GitProvider, error)
-	// TODO: @jpellizzari adding this as a temporary stop-gap to maintain the current behavior for external config repos.
-	// As of https://github.com/weaveworks/weave-gitops/pull/587,
-	// we are not addressing this case yet. Many of the unit tests check for exact function call
-	// arguments, which will get skipped when the AuthService is used (and therefore tests will fail).
-	// Follow up issue where this will be addressed: https://github.com/weaveworks/weave-gitops/issues/592
-	temporaryGitClientFactory func(osysClient osys.Osys, privKeypath string) (git.Git, error)
+	Osys        osys.Osys
+	AppGit      git.Git
+	ConfigGit   git.Git
+	Flux        flux.Flux
+	Kube        kube.Kube
+	Logger      logger.Logger
+	GitProvider gitproviders.GitProvider
```

Comment:
That's an interesting point. I had definitely been viewing the `App` as a disposable entity. I didn't realize the plan was to keep it around. How much of an issue is that from your point of view?

Reply:
I don't think its a huge issue. It's a pretty common pattern to make a new class on every request. Testing becomes harder, however.

This was the reason for the GitProviderFactory pattern we were using before. If "User A" makes a request with their token, we need to create a new GitProvider client with their token, since we don't want "User B" to use the same token.